packages/runtime/src/App.ce.vue

```diff
@@ -21,11 +21,32 @@ const shouldRender = computed(() => {
   return checkerResults.value.some((p) => p.diagnostics.length > 0)
 })
 
-const collapsed = ref<boolean>(!(props?.overlayConfig?.initialIsOpen ?? true))
+// collapsed state if requested (otherwise initially open)
+const initialCollapsed = ref<boolean>(
+  props?.overlayConfig?.initialIsOpen === false || props?.overlayConfig?.initialIsOpen === 'error'
+)
+
+// if initialIsOpen is 'error': watch for results and open if there are errors
+if (props?.overlayConfig?.initialIsOpen === 'error') {
+  // checker result is always initially empty, so need to watch for changes
+  const unwatch = watch(checkerResults, () => {
+    if (!checkerResults.value.length) return
+    // non-zero results length means the badge will appear. decide whether to open the overlay.
+    if (checkerResults.value.some((p) => p.diagnostics.some((d: any) => d.level === 1))) {
+      initialCollapsed.value = false
+    }
+    // stop watching and let the user control the overlay state
+    unwatch()
+  })
+}
+
+const userCollapsed = ref<boolean | undefined>(undefined)
 
 const toggle = () => {
-  collapsed.value = !collapsed.value
+  userCollapsed.value = !(userCollapsed.value ?? initialCollapsed.value)
 }
+
+const collapsed = computed<boolean>(() => userCollapsed.value ?? initialCollapsed.value)
```

Comment:
I considered flipping these, but my lack knowledge of the details of the Vue render cycle gave me pause. 

If the following it better, I'll change it.

```ts
const userCollapsed = ref<boolean | undefined>(undefined)
const collapsed = computed<boolean>(() => userCollapsed.value ?? initialCollapsed.value)

const toggle = () => {
  userCollapsed.value = !collapsed.value
}
```

Reply:
I'd prefer the version in this comment as it's easier to understand. Also, maybe there's a typo in this version?

```diff
- userCollapsed.value = !collapsed.value
+ userCollapsed.value = !userCollapsed.value
```

Reply:
It's intentional. I want to initialize `userCollapsed` based on the current value of `collapsed`. Keep in mind that `userCollapsed` is initially `undefined`, which means `userCollapsed.value = !userCollapsed.value` will _always_ transition from `undefined` to `true`.

Example where that would be a problem:

1. User sets `initialIsOpen: 'error'`.
2. App contains only warnings.
   - Badge will be visible.
   - Overlay will be collapsed.
     - `initialCollapsed = true`
     - `userCollapsed = undefined`
3. User click on badge executing `userCollapsed.value = !userCollapsed.value` will result in `userCollapsed = true`, which means the user doesn't see the overlay appear.

Reply:
To that end, what I currently have in this PR is more verbose, but more clear on intent.

Reply:
@fi3ework Just to confirm before I push again, are you looking for an update to match the comment code?

Reply:
The code looks quite straightforward now! Great work!